File: client-clojure/src/sirqul_io_t_platform/specs/notification_message_list_response.clj
(ns sirqul-io-t-platform.specs.notification-message-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.notification-message-response :refer :all]
            )
  (:import (java.io File)))


(def notification-message-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of notification-message-response-spec)
   (ds/opt :count) int?
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :returning) string?
   })

(def notification-message-list-response-spec
  (ds/spec
    {:name ::notification-message-list-response
     :spec notification-message-list-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/NotificationMessageListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import NotificationMessageListResponse._

case class NotificationMessageListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[NotificationMessageResponse]],
count: Option[Integer],
hasMoreResults: Option[Boolean],
returning: Option[String])

object NotificationMessageListResponse {
  import DateTimeCodecs._

  implicit val NotificationMessageListResponseCodecJson: CodecJson[NotificationMessageListResponse] = CodecJson.derive[NotificationMessageListResponse]
  implicit val NotificationMessageListResponseDecoder: EntityDecoder[NotificationMessageListResponse] = jsonOf[NotificationMessageListResponse]
  implicit val NotificationMessageListResponseEncoder: EntityEncoder[NotificationMessageListResponse] = jsonEncoderOf[NotificationMessageListResponse]
}
